drive-stack/conUDS/src/modules/uds.rs

```diff
@@ -226,34 +227,54 @@ impl UdsClient {
             id[1],
         ];
 
-        info!("Sending ECU read DID command: {:02x?}", buf);
-
         match CanioCmd::send_recv(&buf, self.uds_queue_tx.clone(), 50)
-            .await?
+            .await
+            .map_err(|e| {
+                error!("send_recv failed: {e}");
+                UdsError::GeneralReject
```

Comment:
`GeneralReject` is a very specific UDS response, it shouldn't just be used for a general failure here unless the UDS response was actually `GeneralReject`. Same below